python/lsst/afw/table/schema.py

```diff
@@ -0,0 +1,324 @@
+    Schema._addField_I for 32 bit integers.
+    
+    This method finds the correct type for the current field and calls the appropriate C++ member.
+    """
+    # Check for astropy compatible unit string
+    astropy.units.Unit(units, parse_strict=parse_strict)
+    if type is None:
+        try:
+            prefix, suffix = __builtins__['type'](field).__name__.split("_")
+        except Exception:
+            raise TypeError("First argument to Schema.addField must be a Field if 'type' is not given.")
+        if prefix != "Field":
+            raise TypeError("First argument to Schema.addField must be a Field if 'type' is not given.")
+        attr = "_addField_" + suffix
+        method = getattr(self, attr)
+        return method(field)
+    if not isinstance(type, basestring):
+        type = aliases[type]
+    suffix = _suffixes[type]
+    attr = "_addField_" + suffix
+    method = getattr(self, attr)
+    if size is None:
+        size = globals()["FieldBase_" + suffix]()
+    else:
+        size = globals()["FieldBase_" + suffix](size)
+    return method(field, doc, units, size, doReplace)
+
+def _schemaFind(self, k):
+    """
+    The C++ Schema class has a templated find function, which is exposed to python via
+    Schema._find_Suffix, where suffix is the type exposed to python, for example
+    Schema._find_I for 32 bit integers.
+    
+    This method finds the correct type for the current field and calls the appropriate C++ member.
+    """
+    if not isinstance(k, basestring):
+        
+        try:
+            prefix, suffix = type(k).__name__.split("_")
+        except Exception:
+            raise TypeError("Argument to Schema.find must be a string or Key.")
+        if prefix != "Key":
+            raise TypeError("Argument to Schema.find must be a string or Key.")
+        attr = "_find_" + suffix
+        method = getattr(self, attr)
+        return method(k)
+    for suffix in _suffixes.values():
+        attr = "_find_" + suffix
+        method = getattr(self, attr)
+        try:
+            return method(k)
+        except (lsst.pex.exceptions.TypeError, lsst.pex.exceptions.NotFoundError):
+            pass
+    raise KeyError("Field '%s' not found in Schema." % k)
+
+def _schemaCheckUnits(self, parse_strict='raise'):
+    """
+    Check all of the SchemaItems in a Schema have valid units
+    """
+    for schemaItem in self:
+        astropy.units.Unit(schemaItem.getField().getUnits(), parse_strict=parse_strict)
+
+def _schemaContains(self, key):
+    """
+    Check whether or not a key is found in a Schema.
+    """
+    try:
+        self.find(key)
+        return True
+    except:
+        return False
+
+def schemaComparisonAnd(self, other):
+    """
+    Enumerated types do not have __and__ and __rand__ operators defined, so for
+    SchemaComparisons we cast the Schema.ComparisonFlag enumerated types into integers
+    to do a bitwise comparison.
```

Comment:
Good point, but this is another place mimicking SWIG functionality. So as of now, no one has  needed any other functionality although this could be a new ticket...